examples/mixer_demo.rs

```diff
@@ -30,10 +32,13 @@ fn demo(filename: &Path) {
 
     let frequency = 44100;
     let format = AUDIO_S16LSB; // signed 16 bit samples, in little-endian byte order
-    let channels = 2; // Stereo
+    let channels = DEFAULT_CHANNELS; // Stereo
     let chunk_size = 1024;
     sdl2::mixer::open_audio(frequency, format, channels, chunk_size).unwrap();
-    sdl2::mixer::allocate_channels(0);
+
+    // Number of mixing channels available for sound effect `Chunk`s to play
+    // simultaneously.
+    sdl2::mixer::allocate_channels(4);
```

Comment:
Indeed, it should probably be updated, but I intended to rewrite it completely, and add tons of meaningful comments with it as well (#646). But in the meantime a small comment about what this does would be much needed indeed.